packages/grid/_modules_/grid/components/styled-wrappers/GridRootStyles.ts

```diff
@@ -95,6 +97,7 @@ export const useStyles = makeStyles(
         '& .MuiDataGrid-colCell': {
           position: 'relative',
           display: 'flex',
+          alignItems: 'center',
```

Comment:
have you checked all columns icons alignment with filters sort...
Col can be aligned to the left or right as well

Reply:
I checked it - it seems ok. This is needed in order to keep the col cell separator line centred when the density changes.